extmod/network_wiznet5k.c

```diff
@@ -896,11 +905,11 @@ STATIC mp_obj_t wiznet5k_ifconfig(size_t n_args, const mp_obj_t *args) {
         self->netinfo.dhcp = NETINFO_STATIC;
         mp_obj_t *items;
         mp_obj_get_array_fixed_n(args[1], 4, &items);
-        netutils_parse_ipv4_addr(items[0], netinfo.ip, NETUTILS_BIG);
-        netutils_parse_ipv4_addr(items[1], netinfo.sn, NETUTILS_BIG);
-        netutils_parse_ipv4_addr(items[2], netinfo.gw, NETUTILS_BIG);
-        netutils_parse_ipv4_addr(items[3], netinfo.dns, NETUTILS_BIG);
-        ctlnetwork(CN_SET_NETINFO, &netinfo);
+        netutils_parse_ipv4_addr(items[0], self->netinfo.ip, NETUTILS_BIG);
```

Comment:
Can you explain this change. I think it makes sense, but it would be good to clarify -- is self->netinfo only for DHCP or is it correct to also put the static address into it? What goes wrong if you don't make this change?

Reply:
The existing code doesn't use the IP assigned but it use the IP assigned at the initial.
It is possible to ping test, but when regs() function call, it reads the previous value.

Reply:
I don't understand how this change will affect the regs() function (which reads directly from the wiznet chip).

Reply:
Just ran a test here:

1. Device has default address. Cannot ping.
2. Call nic.ifconfig() with correct static address. Device immediately responds to ping.
3. Call nic.regs(), correctly returns new details.

```
>>> nic.ifconfig(('192.168.1.138', '255.255.255.0', '192.168.1.1', '192.168.1.1'))
>>> nic.regs()
Wiz CREG:
  0000: 00 c0 a8 01 01 ff ff ff 00 02 99 8f 74 61 9a c0
  0010: a8 01 8a 00 00 00 00 00 00 07 d0 08 28 00 00 00
  0020: 00 00 00 00 00 00 ff ff 00 00 00 00 00 00 bf 00
  0030: 00 00 00 00 00 00 00 78 25 04 00 00 00 00 00 00
  0040: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```

So still not sure why this change is necessary?

Reply:
This issue occur the DHCP mode, not Static mode